gcloud-java-core/src/main/java/com/google/gcloud/Service.java

```diff
@@ -16,6 +16,6 @@
 
 package com.google.gcloud;
 
-public interface Service<O extends ServiceOptions> {
-  O options();
+public interface Service<OptionsT extends ServiceOptions<?, OptionsT>> {
+  OptionsT options();
```

Comment:
I think bounding ServiceOptions with generic type breaks BaseService, right?
see Travis build [result] (https://travis-ci.org/GoogleCloudPlatform/gcloud-java/jobs/65651560).
did `mvn clean compile` work for you?